### DISCUSSION TITLE
Avalanche for multivariate streaming sensor data

### DISCUSSION THREAD

**Discussion by Nidarshan619** [ContinualAI/avalanche#417] [Q&A] [answered]
Dear Avalanche team

I am raising this question seeking some assistance in using Avalance tools for signals (instead of image dataset MNIST). 

Please let me explain my problem statement in short through keywords and figures.

- Multivariate streaming sensor data collected for a certain time period representing a disturbance event. 
- Each sensor data when plotted over time represents a disturbance signature. (This problem is very similar to the heart-rate detector or motion sensor identification problem).
- In my problem, I have both scenarios (nc_scenario and ni _scenario)
1. nc_scenario – there may be a new class of disturbance signatures recorded over time.
2. ni_scenario – New instances are recorded due to concept drift of the existing class of disturbance signatures.
-  Each cube in the ‘below figure’ shall be updated in an incremental fashion to the deep learning algorithm.
 ![Dataset representation resize](https://user-images.githubusercontent.com/55737586/111303635-063dfd00-8655-11eb-98b0-d0b69d5595e0.png)
I find it difficult to understand how I replace the image dataset from your examples with the sequential data described above. 

Thank you in advance!

Best Regards,
Nidarshan Kumar

**Comment by AndreaCossu** [comment, marked as answer]
Hi @Nidarshan619 ! Adding support for time series data is something we are working on. However, under certain conditions time series data could already be directly integrated in Avalanche.

1. Are your time series all of the same lengths (or can you pad them accordingly?)
2. Is your problem a sequence classification? That is, you classify the entire sequence, you don't want to emit a prediction on every time step.

If you met these two conditions you should create a new `Dataset` in Avalanche and then pass it to the `nc_scenario` or `ni_scenario` to get back a continual learning scenario.

**Reply by vlomonaco** [reply]
@Nidarshan619 does the @AndreaCossu's post answers your question? :)

**Reply by Nidarshan619** [reply]
Yes, thank you very much, Andrea and Vlomonaco
